### ISSUE TITLE
[Maps] [Android] [Regression] Registering for CameraMove events in a custom MapHandler disables Map.VisibleRegion.Center

### ISSUE THREAD

**Issue by mfeingol** [dotnet/maui#21094]
### Description

Create a custom map class and a handler for it, and inside that handler register for CameraMove events from the underlying GoogleMap (either via SetOnCameraMoveListener or the CameraMove event). When you do this, the Maui Map's VisibleRegion property no longer updates when you move the map. In particular, the Center property always returns the map's initial location.

Comment out the += to the CameraMove event, or the call to SetOnCameraMoveListener, and VisibleRegion.Center will update again. This is very broken, and it is a regression from Xamarin.Forms.

cf https://github.com/dotnet/maui/issues/13186 - I mistakenly closed this thinking the issue was fixed. It was not.

### Steps to Reproduce

```c#
class CustomMapHandler : MapHandler
{
    protected override void ConnectHandler(Android.Gms.Maps.MapView platformView)
    {
        base.ConnectHandler(platformView);
        platformView.GetMapAsync(new CustomMapCallbackHandler(this));
    }

    internal void OnMapReady(Android.Gms.Maps.GoogleMap googleMap)
    {
        googleMap.CameraMove += this.OnCameraMoved;
    }

    void OnCameraMoved(object sender, EventArgs e)
    {
    }
}

class CustomMapCallbackHandler : Java.Lang.Object, Android.Gms.Maps.IOnMapReadyCallback
{
    readonly CustomMapHandler handler;

    public CustomMapCallbackHandler(CustomMapHandler handler)
    {
        this.handler = handler;
    }

    public void OnMapReady(Android.Gms.Maps.GoogleMap map)
    {
        this.handler.OnMapReady(map);
    }
}
```

### Link to public reproduction project repository

https://github.com/mfeingol/repros/tree/master/MapVisibleRegionRepro

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

14

### Did you find any workaround?

No workaround.

**Comment by mfeingol** [comment]
Hello. There's a repro mentioned above: https://github.com/mfeingol/repros/tree/master/MapVisibleRegionRepro

**Comment by NirmalKumarYuvaraj** [comment]
@mfeingol , This occurs because Android's GoogleMap only supports one camera move listener at a time. When custom handlers call SetOnCameraMoveListener(), they completely replace MAUI's internal MapCallbackHandler that is responsible for updating the visible region. This replacement breaks the framework's ability to track map state changes. If you are replacing the existing handler with custom logic, you must implement the logic to handle the visible region updates yourself, including calling the appropriate methods to ensure the VisibleRegion.Center property remains accurate. 

https://github.com/dotnet/maui/blob/c65f4338790fe79ab68df5551a0797ccf4c18bda/src/Core/maps/src/Handlers/Map/MapHandler.Android.cs#L302-L317

**Comment by mfeingol** [comment]
Yep, I figured that out while writing my own map control on top of the Android Google Map. Thanks for the note.
